packages/ag-charts-enterprise/src/series/map-marker/mapMarkerSeries.ts

```diff
@@ -530,8 +530,17 @@ export class MapMarkerSeries
             _ModuleSupport.PlacedLabel<_ModuleSupport.PointLabelDatum>
         >;
     }) {
-        const placedLabels = (this.isLabelEnabled() ? this.chart?.placeLabels().get(this) : undefined) ?? [];
-        return opts.labelSelection.update(placedLabels);
+        if (!this.isLabelEnabled()) return opts.labelSelection.update([]);
+
+        this.ctx.seriesLabelLayoutManager
+            .placeLabels(this.id, this.getLabelData())
+            .then((layoutResult) => {
+                const placedLabels = layoutResult.get(this.id) ?? [];
+                opts.labelSelection.update(placedLabels);
+                this.updateLabelNodes(opts);
+            })
+            .catch((e) => Logger.error(e));
+        return opts.labelSelection;
```

Comment:
Shouldn't we wait for the placement to finish. before returning `labelSelection`? (which isn't yet updated)

Reply:
Technically yes, but practically no.

Without a lot of additional change to the rest of the `chart.update()` flow to support async execution, we can't 'wait' here. I've checked the control flow, and calling `this.updateLabelNodes()` is the scope of effects after the placed data is calculated.

The new `Chart.updateSeries()` methods coordinates with `SeriesLabelLayoutManager` to ensure these pending updates happen before the main process flow continues.

Practically we need to rework the entire label handling process, that is too large to address in this tactical PR.

Reply:
Discussed offline, re-arranged the processing so that we aren't doing async processing, and the flow is a bit clearer.